src/main/java/com/smartsheet/api/RowAttachmentResources.java

```diff
@@ -0,0 +1,103 @@
+package com.smartsheet.api;
+
+/*
+ * #[license]
+ * Smartsheet SDK for Java
+ * %%
+ * Copyright (C) 2014 Smartsheet
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * %[license]
+ */
+
+import com.smartsheet.api.models.Attachment;
+import com.smartsheet.api.models.DataWrapper;
+import com.smartsheet.api.models.PaginationParameters;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ * <p>This interface provides methods to access RowAttachment resources.</p>
+ *
+ * <p>Thread Safety: Implementation of this interface must be thread safe.</p>
```

Comment:
Not sure I agree with this assertion, since we've seen issues in the Java backup utility where it was assumed that the sdk was thread safe when in fact it is not, particularly if you are changing the Assume User value or the access token....  Furthermore, you're not going to ever create a single instance of the SDK in a web app and have all requests funnel through it, since the instance of Smartsheet is user-specific. Not a big deal, just seems overly restricting.